benchmarks/domains/synthetic_2C1D_1C.py

```diff
@@ -0,0 +1,125 @@
+from baybe.parameters import NumericalContinuousParameter, NumericalDiscreteParameter
+from baybe.recommenders.pure.nonpredictive.sampling import RandomRecommender
+from baybe.searchspace import SearchSpace
+from baybe.simulation import simulate_scenarios
+from baybe.targets import NumericalTarget, TargetMode
+from benchmarks.definition import Benchmark
+from benchmarks.definition.config import (
+    DEFAULT_RECOMMENDER,
+    RecommenderConvergenceAnalysis,
+)
+
+if TYPE_CHECKING:
+    from mpl_toolkits.mplot3d import Axes3D
+
+
+def lookup(z: np.ndarray, x: np.ndarray, y: np.ndarray) -> np.ndarray:
+    """Lookup function.
+
+    Inputs:
+        z   discrete   {1,2,3,4}
+        x   continuous [-2*pi, 2*pi]
+        y   continuous [-2*pi, 2*pi]
+    Output: continuous
+    Objective: Maximization
+    Optimal Inputs:
+        {x: 1.610, y: 1.571, z: 3}
+        {x: 1.610, y: -4.712, z: 3}
+    Optimal Output: 4.09685
+    """
+    try:
+        assert np.all(-2 * pi <= x) and np.all(x <= 2 * pi)
+        assert np.all(-2 * pi <= y) and np.all(y <= 2 * pi)
+        assert np.all(np.isin(z, [1, 2, 3, 4]))
+    except AssertionError:
+        raise ValueError("Inputs are not in the valid ranges.")
+
+    return (
+        (z == 1) * sin(x) * (1 + sin(y))
+        + (z == 2) * (x * sin(0.9 * x) + sin(x) * sin(y))
+        + (z == 3) * (sqrt(x + 8) * sin(x) + sin(x) * sin(y))
+        + (z == 4) * (x * sin(1.666 * sqrt(x + 8)) + sin(x) * sin(y))
+    )
+
+
+def benchmark_callable(scenario_config: RecommenderConvergenceAnalysis) -> DataFrame:
+    """Run different recommenders against the test function."""
+    parameters = [
+        NumericalContinuousParameter("x", (-2 * pi, 2 * pi)),
+        NumericalContinuousParameter("y", (-2 * pi, 2 * pi)),
+        NumericalDiscreteParameter("z", (1, 2, 3, 4)),
+    ]
+
+    objective = NumericalTarget(name="target", mode=TargetMode.MAX).to_objective()
+
+    scenarios: dict[str, Campaign] = {}
+    for scenario_name, recommender in scenario_config.recommenders.items():
+        recommender_dct = (
+            {} if recommender is DEFAULT_RECOMMENDER else {"recommender": recommender}
+        )
+        campaign = Campaign(
+            searchspace=SearchSpace.from_product(parameters=parameters),
+            objective=objective,
+            **recommender_dct,
+        )
+        scenarios[scenario_name] = campaign
+
+    return simulate_scenarios(
+        scenarios,
+        lookup,
+        batch_size=scenario_config.batch_size,
+        n_doe_iterations=scenario_config.n_doe_iterations,
+        n_mc_iterations=scenario_config.n_mc_iterations,
+        impute_mode="error",
+    )
+
+
+benchmark_config = RecommenderConvergenceAnalysis(
+    recommenders={
+        "Default Recommender": DEFAULT_RECOMMENDER,
+        "Random Recommender": RandomRecommender(),
+    },
+    batch_size=5,
+    n_doe_iterations=30,
+    n_mc_iterations=50,
+)
+
+benchmark = Benchmark(
+    name="synthetic_2C1D_1C",
```

Comment:
@AVHopp @Scienfitz: not sure what is a good naming convention here. So far, I've simply represented type and inputs/outputs. However, I can already see that we might have another synthetic function of this kind + we might kick some benchmarks in the future. If we simply added numbers, this would bring us situation that our benchmarks will at some point start like `synthetic4...` but there is no `synthetic1...`. Not sure if that is confusing or not. Alternatively, we could:
* use a random suffix, like azure does it for PRs, i.e. `synthetic4823_2C1D_1C`
* or use a completely generated human-readable name like `silver-unicorn`

I also see the problem that the name currently does not convey that it is a "convergence-type" benchmark. Ideas?

Reply:
Maybe one more thought: consider that most of the times the names will be used by us in meetings to reference a particular benchmark verbally. So the `silver-unicorn` type of naming might actually not be a bad idea – it's catchy and easy to pronounce. Or to put it in [xkcd](https://xkcd.com/936/) terms: correct horse battery staple

Reply:
Let's think about what potential benchmarking functions could look like:

1. They can either be `analytic` in the way that we can actually write down the function or `heuristic` in the sense that one cannot write down the optimized function.
2. Each function has a fixed number of inputs and outputs
3. However, I really like the catchy names :D

So I'd propose the scheme `{catchy adjective}_{catchy animal}_{ana/heu}_{Inputs}_{Outputs}`, so this function could be `silver_unicorn_ana_2C1D_1C`. This has some redundancy, but I guess this does not hurt.

Reply:
ngl, this feels way to overconstructed, `synthetic1` or `analytic2` etc are equally human readable, random or animal names are just silly, have fun explain the name on a slide displaying `uniorn-rocket_2d1c_2t`.

We construct these, so why do you need randomization? And why on earth is this ` If we simply added numbers, this would bring us situation that our benchmarks will at some point start like synthetic4... but there is no synthetic1.... Not sure if that is confusing or not.` your base case?? We can simply add this benchmark and call it `1` and increase with every upcoming one. We can always kick the from the `BENCHMARKS` list without causing any hole in the enumeration